src/cpp/transport/discovery/DiscoveryServerWindows.cpp

```diff
@@ -0,0 +1,154 @@
+        {
+            /* Get local address. */
+            SOCKET fd = socket(PF_INET, SOCK_DGRAM, 0);
+            struct sockaddr_in temp_addr;
+            temp_addr.sin_family = AF_INET;
+            temp_addr.sin_port = htons(80);
+            temp_addr.sin_addr.s_addr = inet_addr("1.2.3.4");
+            int connected = connect(fd, (struct sockaddr *)&temp_addr, sizeof(temp_addr));
+            if (0 == connected)
+            {
+                struct sockaddr local_addr;
+                int local_addr_len = sizeof(local_addr);
+                if (-1 != getsockname(fd, &local_addr, &local_addr_len))
+                {
+                    transport_address_.medium_locator().address({uint8_t(local_addr.sa_data[2]),
+                                                                 uint8_t(local_addr.sa_data[3]),
+                                                                 uint8_t(local_addr.sa_data[4]),
+                                                                 uint8_t(local_addr.sa_data[5])});
+                    rv = true;
+                }
+                closesocket(fd);
+            }
+        }
+    }
+
+    return rv;
+}
+
+bool DiscoveryServerWindows::close()
+{
+    return (0 == closesocket(poll_fd_.fd));
+}
+
+bool DiscoveryServerWindows::recv_message(InputPacket& input_packet, int timeout)
+{
+    bool rv = true;
+    struct sockaddr client_addr;
+    int client_addr_len = sizeof(client_addr);
+
+    int poll_rv = WSAPoll(&poll_fd_, 1, timeout);
+    if (0 < poll_rv)
+    {
+        int bytes_received = recvfrom(poll_fd_.fd, (char*)buffer_, sizeof(buffer_), 0, &client_addr, &client_addr_len);
+        if (SOCKET_ERROR != bytes_received)
+        {
+            input_packet.message.reset(new InputMessage(buffer_, static_cast<size_t>(bytes_received)));
+            uint32_t addr = ((struct sockaddr_in*)&client_addr)->sin_addr.s_addr;
+            uint16_t port = ((struct sockaddr_in*)&client_addr)->sin_port;
+            input_packet.source.reset(new UDPEndPoint(addr, port));
+        }
+    }
+    else
+    {
+        rv = false;
+        if (0 == poll_rv)
+        {
+            errno = ETIME;
+        }
+    }
+
+    return rv;
+}
+
+bool DiscoveryServerWindows::send_message(OutputPacket output_packet)
+{
+    bool rv = true;
+    const UDPEndPoint* destination = static_cast<const UDPEndPoint*>(output_packet.destination.get());
+    struct sockaddr_in client_addr;
+
+    client_addr.sin_family = AF_INET;
+    client_addr.sin_port = destination->get_port();
+    client_addr.sin_addr.s_addr = destination->get_addr();
+    int bytes_sent = sendto(poll_fd_.fd,
+                            (char*)output_packet.message->get_buf(),
+                            output_packet.message->get_len(),
+                            0,
+                            (struct sockaddr*)&client_addr,
+                            sizeof(client_addr));
+    if (SOCKET_ERROR != bytes_sent)
```

Comment:
Initializing the `rv` to *false*, I think, this code could be more readeable:
```c
if ((SOCKET_ERROR != bytes_sent) && ((size_t)bytes_sent == output_packet.message->get_len()))
{
    rv = true;
}

Reply:
The same as before. Done!